File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task191_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Solution {
    public int longestValidParentheses(String s) {
        int max = 0;
        int lastInvalid = -1;
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                stack.push(i);
            } else {
                if (stack.isEmpty()) {
                    lastInvalid = i;
                } else {
                    stack.pop();
                    if (stack.isEmpty()) {
                        max = Math.max(max, i - lastInvalid);
                    } else {
                        max = Math.max(max, i - stack.peek());
                    }
                }
            }
        }
        return max;
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task191_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <string.h>
#include <stdio.h>

int longestValidParentheses(char* s) {
    int max = 0;
    int lastInvalid = -1;
    int stack[10000];
    int top = -1;
    for (int i = 0; i < strlen(s); i++) {
        if (s[i] == '(') {
            stack[++top] = i;
        } else {
            if (top == -1) {
                lastInvalid = i;
            } else {
                top--;
                if (top == -1) {
                    max = max > i - lastInvalid ? max : i - lastInvalid;
                } else {
                    max = max > i - stack[top] ? max : i - stack[top];
                }
            }
        }
    }
    return max;
}